src/pools/gyro2Pool/gyro2Pool.ts

```diff
@@ -160,20 +161,42 @@ export class Gyro2Pool implements PoolBase {
     }
 
     getLimitAmountSwap(
-        poolPairData: PoolPairBase,
+        poolPairData: Gyro2PoolPairData,
         swapType: SwapTypes
     ): OldBigNumber {
         if (swapType === SwapTypes.SwapExactIn) {
+            const balances = [poolPairData.balanceIn, poolPairData.balanceOut];
+            const normalizedBalances = _normalizeBalances(
+                balances,
+                poolPairData.decimalsIn,
+                poolPairData.decimalsOut
+            );
+            const invariant = _calculateInvariant(
+                normalizedBalances,
+                poolPairData.sqrtAlpha,
+                poolPairData.sqrtBeta
+            );
+            const maxAmountInAssetInPool = mulDown(
+                invariant,
+                poolPairData.sqrtBeta.sub(poolPairData.sqrtAlpha)
```

Comment:
This looks wrong. Assume `!tokenInIsToken0` like in the writeup. Then `poolPairData.sqrtAlpha == 1/gyro2Pool.sqrtBeta` and `poolPairData.sqrtBeta == 1/gyro2Pool.sqrtAlpha`. Then  `maxAmountInAssetInPool == invariant * (1/gyro2Pool.sqrtAlpha - 1/gyro2Pool.sqrtBeta)`. But in the writeup we want `maxAmountInAssetInPool  == y+ = invariant * (poolPairData.sqrtBeta - gyro2Pool.sqrtAlpha)`.
I think you want the formula for x+ from the writeup?